resource/resource_test.go

```diff
@@ -134,53 +126,63 @@ func TestOut(t *testing.T) {
 	type want struct {
 		version  oc.Version
 		metadata oc.Metadata
+		body     map[string]string
 		err      error
 	}
 	var testCases = []struct {
 		name string
 		in   in
 		want want
 	}{
-		{
-			"valid mandatory sources",
+		{"valid mandatory sources",
 			in{defSource, defParams, defEnv},
-			want{defVersion, defMeta, nil},
+			want{defVersion, defMeta, nil, nil},
 		},
-		{
-			"missing mandatory sources",
+		{"missing mandatory sources",
 			in{oc.Source{}, defParams, defEnv},
-			want{nil, nil, &missingSourceError{}},
+			want{nil, nil, nil, &missingSourceError{}},
 		},
-		{
-			"unknown source",
+		{"unknown source",
 			in{oc.Source{"access_token": "x", "owner": "a", "repo": "b", "pizza": "napoli"},
 				defParams, defEnv},
-			want{nil, nil, &unknownSourceError{}},
+			want{nil, nil, nil, &unknownSourceError{}},
 		},
-		{
-			"valid mandatory parameters",
+		{"valid mandatory parameters",
 			in{defSource, defParams, defEnv},
-			want{defVersion, defMeta, nil},
+			want{defVersion, defMeta, nil, nil},
 		},
-		{
-			"completely missing mandatory parameters",
+		{"completely missing mandatory parameters",
 			in{defSource, oc.Params{}, defEnv},
-			want{nil, nil, &missingParamError{}},
+			want{nil, nil, nil, &missingParamError{}},
 		},
-		{
-			"invalid state parameter",
+		{"invalid state parameter",
 			in{defSource, oc.Params{"state": "hello"}, defEnv},
-			want{nil, nil, &invalidParamError{}},
+			want{nil, nil, nil, &invalidParamError{}},
 		},
-		{
-			"unknown parameter",
+		{"unknown parameter",
 			in{defSource, oc.Params{"state": "pending", "pizza": "margherita"}, defEnv},
-			want{nil, nil, &unknownParamError{}},
+			want{nil, nil, nil, &unknownParamError{}},
 		},
-		{
-			"do not return a nil version the first time it runs (see Concourse PR #4442)",
+		{"do not return a nil version the first time it runs (see Concourse PR #4442)",
 			in{defSource, defParams, defEnv},
-			want{defVersion, defMeta, nil},
+			want{defVersion, defMeta, nil, nil},
+		},
+		{"source: optional: context_prefix",
+			in{
+				oc.Source{
+					"access_token":   cfg.Token,
+					"owner":          cfg.Owner,
+					"repo":           cfg.Repo,
+					"context_prefix": "cocco"},
+				defParams,
+				defEnv,
+			},
+			want{
+				defVersion,
+				defMeta,
+				map[string]string{"context": "cocco/" + defEnv.Get("BUILD_JOB_NAME")},
```

Comment:
This is what we expect from the test. Since the HTTP payload is JSON encoded, we don't want to pass the full JSON object, we want to pass only what we actually want to test. The Go json package allows to use a map, which is perfect for this use (see also comments below)